src/component/FetchItems.js

```diff
@@ -7,17 +7,61 @@ import DeleteToken from './DeleteToken';
 const FetchItems = ({ token, setToken, firestore }) => {
   const [empty, setEmpty] = useState(true);
 
+  // stores the number of milliseconds elapsed since January 1, 1970
+  const todayInMs = Date.now();
+  const today = new Date();
+
+  const itemsDocRef = firestore
+    .collection('lists')
+    .doc(token)
+    .collection('items');
+
+  // function to change database on button click
+  const handlePurchase = event => {
+    let itemId = event.target.id;
+    event.preventDefault();
+    itemsDocRef
+      .doc(itemId)
+      .get()
+      .then(updateDatabase);
+  };
+
+  const updateDatabase = data => {
+    itemsDocRef.doc(data.id).update({
+      purchasedWithinLastDay: true,
```

Comment:
As [I mentioned in Slack](https://the-collab-lab.slack.com/archives/CPPP4NKE3/p1576435576009600?thread_ts=1576435127.009500&cid=CPPP4NKE3), we should compute this value when rendering the list of items rather than storing it in the database.

It is generally good practice to calculate what you can rather than storing it, but in this case the data itself—specifically `purchasedWithinLastDay `—will quickly be out of date. Keeping it up-to-date is possible, but not easy and would be error prone/one more moving part that could have bugs.

Reply:
@segdeha I realized we don't even need the purchasedWithinLastDay value. We can simply use the time difference logic to update the class names. So I removed that altogether!

Reply:
Perfect!